api/src/routers/reservations.js

```diff
@@ -0,0 +1,80 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const reservRouter = express.Router();
+// get all meal
+reservRouter.get("/", async (req, res, next) => {
+    try{
+        const all_reservation = await knex("Reservation").select("*");
+    res.json(all_reservation);
+    } catch (err) {
+        next(err);
+      }
+});
+
+reservRouter.post("/", async (req, res, next) => {
+    try{
+        const data=req.body;
+        await knex("Reservation").insert(data);
+        res.status(200).json({ message: "created successfully" });
+    } catch (err) {
+        next(err);
+      }
+});
+
+reservRouter.get("/:id", async (req,res,next) => {
+    try{
+        const id = req.params.id;
+        const reservation = await knex("Reservation").select("*").where("id",id).first();
+        if (!reservation) {
+            res.status(404).json({ message: "reservation not found" });
+          } else {
+            res.json(reservation);
+          }
+    } catch (err) {
+        next(err);
+      }
+});
+
+reservRouter.put("/:id", async (req, res, next) => {
+    const { id } = req.params;
+    const updatedData = req.body;
+  
+    try {
+        const result = await knex("Reservation")
+            .where({ id })
+            .update(updatedData);
+  
+        if (result) {
+          const updatedReservation = await knex("Reservation").where({ id }).first();
```

Comment:
you are using `where(...)` in two ways, here in the `put` endpoint and in the above `get` endpoint. Choose one way and be consistent